vms/platformvm/state/metadata_delegator_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+)
+
+func TestParseDelegatorMetadata(t *testing.T) {
+	type test struct {
+		name        string
+		bytes       []byte
+		expected    *delegatorMetadata
+		expectedErr error
+	}
+	tests := []test{
+		{
+			name: "potential reward only no codec",
+			bytes: []byte{
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x7b,
+			},
+			expected: &delegatorMetadata{
+				PotentialReward: 123,
+				StakerStartTime: 0,
+			},
+			expectedErr: nil,
+		},
+		{
```

Comment:
this should really not be happening in prod, but showcase that codec works on v0 differently than no-codec does

Reply:
Removed this test and updated delegator metadata to only marshal on codec version 1